ncs/Kconfig

```
@@ -59,3 +59,51 @@ config SUIT_DFU_CACHE_EXTRACT_IMAGE_URI
 	default "cache://rad_recovery.bin" if (SOC_NRF54H20_CPURAD_COMMON || SOC_NRF9230_ENGB_CPURAD) && SUIT_RECOVERY
 
 endif # SUIT_DFU_CACHE_EXTRACT_IMAGE
+
+config SUIT_ENVELOPE_TARGET_ENCRYPT
+	bool "Encrypt the target image"
+
+if SUIT_ENVELOPE_TARGET_ENCRYPT
+
+config SUIT_ENVELOPE_TARGET_ENCRYPT_STRING_KEY_ID
+	string "The string key ID used to identify the encryption key on the device"
+	default "FWENC_APPLICATION_GEN1" if SOC_NRF54H20_CPUAPP_COMMON
+	default "FWENC_RADIOCORE_GEN1" if SOC_NRF54H20_CPURAD_COMMON
+	help
+	  This string is translated to the numeric KEY ID by the encryption script
+
+config SUIT_ENVELOPE_TARGET_ENCRYPT_KEY_NAME
+	string "Name of the key used for encryption - to identify the key in the KMS"
+	default SUIT_ENVELOPE_TARGET_ENCRYPT_STRING_KEY_ID
+
+
+choice SUIT_ENVELOPE_TARGET_ENCRYPT_PLAINTEXT_HASH_ALG
+	prompt "Algorithm used to calculate the digest of the plaintext firmware"
+	default SUIT_ENVELOPE_TARGET_ENCRYPT_PLAINTEXT_HASH_ALG_SHA256
+
+config SUIT_ENVELOPE_TARGET_ENCRYPT_PLAINTEXT_HASH_ALG_SHA256
+	bool "Use the SHA-256 algorithm"
+
+config SUIT_ENVELOPE_TARGET_ENCRYPT_PLAINTEXT_HASH_ALG_SHA384
+	bool "Use the SHA-384 algorithm"
+
+config SUIT_ENVELOPE_TARGET_ENCRYPT_PLAINTEXT_HASH_ALG_SHA512
+	bool "Use the SHA-512 algorithm"
+
+config SUIT_ENVELOPE_TARGET_ENCRYPT_PLAINTEXT_HASH_ALG_SHAKE128
+	bool "Use the SHAKE128 algorithm"
+
+config SUIT_ENVELOPE_TARGET_ENCRYPT_PLAINTEXT_HASH_ALG_SHAKE256
+	bool "Use the SHAKE256 algorithm"
+
+endchoice
+
+config SUIT_ENVELOPE_TARGET_ENCRYPT_PLAINTEXT_HASH_ALG_NAME
+	string "String name of the algorithm used to calculate the digest of the plaintext firmware"
```

```suggestion
	string
```
Should not have a prompt